src/db.rs

```diff
@@ -323,3 +321,1027 @@ pub async fn dependencies_by_dataset_select(
 
     Ok(dependencies)
 }
+
+#[cfg(test)]
```

Comment:
@coderabbitai please create a PR that applies DRY principles to the unit test in `db.rs` so we don't need to repeat our selves over and over with creating datasets, data products, and dependencies. We should create common function and use those functions in the unit test to reduce the leaves of boilerplate code.